database/db.go

```diff
@@ -0,0 +1,108 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/TanyaEIEI/pokedex/graph/model"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+type Pokebase struct {
+	Db *gorm.DB
+}
+
+func InitDb() (*gorm.DB, error) {
+	db, err := gorm.Open(sqlite.Open("testdb1.db"), &gorm.Config{
+		Logger: logger.Default.LogMode(logger.Silent),
+	})
+	if err != nil {
+		return nil, err
+	}
+	db.AutoMigrate(&Pokemon{})
+	return db, nil
+}
+
+func (pkb *Pokebase) CreatePokemon(ctx context.Context, input Pokemon) (*Pokemon, error) {
+
+	// Convert input from GraphQL model to database model
+	newCharacter := Pokemon{
+		Name:        input.Name,
+		Description: input.Description,
+		Category:    input.Category,
+		Abilities:   input.Abilities,
+		Type:        input.Type,
+	}
+	if err := pkb.Db.Create(&newCharacter).Error; err != nil {
+		fmt.Printf("Failed to create Pokemon: %v", err)
+		return nil, fmt.Errorf("failed to create Pokemon")
+	}
+
+	return &newCharacter, nil
+}
+
+func (pkb *Pokebase) ListPokemon(ctx context.Context) ([]*model.Pokemon, error) {
+	var pokemonList []*model.Pokemon
+	err := pkb.Db.Find(&pokemonList).Error
+	if err != nil {
+		return nil, err
+	}
+	return pokemonList, nil
+}
+
+func (pkb *Pokebase) UpdatePokemon(ctx context.Context, input UpdatePokemonInput) (*Pokemon, error) {
+	var pokemonData *Pokemon
+	if err := pkb.Db.First(&pokemonData, input.ID).Error; err != nil {
+		fmt.Printf("Failed to find Pokemon for update: %v", err)
+		return nil, fmt.Errorf("failed to find Pokemon for update")
+	}
+
+	pokemonData.Name = *input.Name
+	pokemonData.Description = *input.Description
+	pokemonData.Category = *input.Category
+	pokemonData.Abilities = *input.Abilities
+	pokemonData.Type = *input.Type
```

Comment:
what happen if input.Name, input.Description, input.Category, input.Abilities is null ?

Reply:
i added more conditions in schema.reslover.go on 'CreatePokemon' & 'UpdatePokemon' Func kub

Reply:
additional explanation: i check data is nil or not before db function process kub ,